server/cluster.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"database/sql"
+
+	"github.com/mattermost/squirrel"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func pingClusterDiscoveryTable(db *sql.DB, driverName, clusterName string) ([]*model.ClusterDiscovery, error) {
+	builder := squirrel.StatementBuilder.PlaceholderFormat(getQueryPlaceholder(driverName))
+
+	query := builder.Select("*").From("ClusterDiscovery").
+		Where(squirrel.Eq{"Type": model.CDSTypeApp}).
+		Where(squirrel.Eq{"ClusterName": clusterName}).
+		Where(squirrel.Gt{"LastPingAt": model.GetMillis() - model.CDSOfflineAfterMillis})
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "cluster_discovery_tosql")
+	}
+
+	rows, err := db.Query(queryString, args...)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find ClusterDiscovery")
+	}
+	defer rows.Close()
+
+	list := []*model.ClusterDiscovery{}
+	if err := rows.Scan(&list); err != nil {
+		return nil, err
+	}
+
+	return list, nil
+}
+
+func getQueryPlaceholder(driverName string) squirrel.PlaceholderFormat {
+	if driverName == model.DatabaseDriverPostgres {
+		return squirrel.Dollar
+	}
+	return squirrel.Question
+}
```

Comment:
Let's move this to a separate file. It's unfortunate though. There's a lot of copy-paste from the server code.